aws-api/src/main/java/com/amplifyframework/api/aws/GsonGraphQLResponseFactory.java

```diff
@@ -31,24 +34,30 @@
 import java.lang.reflect.Type;
 import java.util.ArrayList;
 import java.util.Collections;
+import java.util.Date;
 import java.util.List;
 
 /**
  * Converts JSON strings into models of a given type, using Gson.
  */
 final class GsonGraphQLResponseFactory implements GraphQLResponse.Factory {
+    private static final String DATA_KEY = "data";
+    private static final String ERRORS_KEY = "errors";
+    private static final String ITEMS_KEY = "items";
+
     private final Gson gson;
 
     /**
      * Default constructor using default Gson object.
      */
     GsonGraphQLResponseFactory() {
-        this(
-                new GsonBuilder()
-                .registerTypeAdapter(List.class, new GsonListDeserializer())
```

Comment:
Don't you need the `List.class` binding anymore?

Reply:
As far as I can tell, `GsonListDeserializer` isn't actually used by this factory.  It looks like it was designed mainly to skip the "items" level when deserializing an array, but it turns out this is [handled manually in GsonGraphQLResponseFactory](https://github.com/aws-amplify/amplify-android/blob/c845cf161f4af41461fdc58a8f3686aee6ae8502/aws-api/src/main/java/com/amplifyframework/api/aws/GsonGraphQLResponseFactory.java#L132-L133). 

Of note though, I am just deleting the one in aws-api.  There is a [copy of GsonListDeserializer in aws-datastore](https://github.com/aws-amplify/amplify-android/blob/2922f3ce35d26a420892b46694dec231df75c3b0/aws-datastore/src/main/java/com/amplifyframework/datastore/appsync/AppSyncResponseDeserializer.java#L85), and I have not checked to see if that one is used or not.